File: sdk/python/src/agentcc/types/chat/chat_completion.py
"""ChatCompletion — the primary response model for chat completions."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, ConfigDict

from agentcc.types.chat.chat_completion_message import ChatCompletionMessage
from agentcc.types.agentcc_metadata import AgentCCMetadata
from agentcc.types.shared import Usage


class Choice(BaseModel):
    model_config = ConfigDict(extra="allow")

    index: int
    message: ChatCompletionMessage
    finish_reason: str | None = None
    logprobs: Any = None


class ChatCompletion(BaseModel):
    model_config = ConfigDict(extra="allow")

    id: str
    object: str
    created: int
    model: str
    choices: list[Choice]
    usage: Usage | None = None
    system_fingerprint: str | None = None
    service_tier: str | None = None
    agentcc: AgentCCMetadata | None = None

File: sdk/typescript/src/types/chat/chat-completion.ts
import type { Usage, AgentCCMetadata } from "../shared.js";

export interface FunctionCall {
  name: string;
  arguments: string;
}

export interface ToolCall {
  id: string;
  type: "function";
  function: FunctionCall;
}

export interface ChatCompletionMessage {
  role: "system" | "user" | "assistant" | "tool" | "function";
  content: string | null;
  function_call?: FunctionCall;
  tool_calls?: ToolCall[];
  name?: string;
  refusal?: string | null;
}

export interface Choice {
  index: number;
  message: ChatCompletionMessage;
  finish_reason: string | null;
  logprobs?: unknown;
}

export interface ChatCompletion {
  id: string;
  object: "chat.completion";
  created: number;
  model: string;
  choices: Choice[];
  usage?: Usage;
  system_fingerprint?: string;
  service_tier?: string;
  /** Gateway metadata — injected by the SDK. */
  agentcc?: AgentCCMetadata;
  [key: string]: unknown;
}
